src/ToolBox/superpmi/superpmi-shared/lightweightmap.h

```diff
@@ -369,23 +371,37 @@ class LightWeightMap : public LightWeightMapBuffer
     int GetIndex(_Key key)
     {
         if (numItems == 0)
+        {
             return -1;
+        }
 
-        // O(log n) version
-        int first = 0;
-        int mid   = 0;
-        int last  = numItems;
```

Comment:
Couldn't you just fix this bug by changing this to `int last = numItems - 1`, without all the rest of the code churn?

Reply:
Nop, then you will have a problem with search in 2 element array, `last == 2 - 1 == 1, first == 0, mid == (1 + 0) / 2 == 0`, so [1] would not be checked.

Reply:
The implementation when one border is included `[` and the other is excluded `)` and the loop check is (right - left > 1) has the least number or corner cases that you have to check inside the loop.

Reply:
I was not able to find a bad case. With the 'key==elem[mid]` check inside the loop it works, but without it the loop `while (first <= last)` will be infinite with such borders.

The PR was updated.